pkg/generator.go

```diff
@@ -329,11 +380,12 @@ type namer interface {
 func (g *Generator) renderType(ctx context.Context, typ types.Type) string {
 	switch t := typ.(type) {
 	case *types.Named:
-		o := t.Obj()
-		if o.Pkg() == nil || o.Pkg().Name() == "main" || (!g.KeepTree && g.InPackage && o.Pkg() == g.iface.Pkg) {
-			return o.Name()
+		return g.addPackageScopedType(ctx, t.Obj())
+	case *types.TypeParam:
+		if t.Constraint() != nil {
+			return t.Obj().Name()
 		}
-		return g.addPackageImport(ctx, o.Pkg()) + "." + o.Name()
+		return g.addPackageScopedType(ctx, t.Obj())
```

Comment:
Can you explain the difference between the `t.Constraint() != nil` block and the other block?